### ISSUE TITLE
F-Droid can't build

### ISSUE THREAD

**Issue by licaon-kter** [privacybydesign/irmamobile#163]
ref: https://gitlab.com/fdroid/fdroiddata/-/jobs/4053212260#L788

/LE: fyi https://gitlab.com/fdroid/fdroiddata/-/commit/324469753aeca008a413f062a32cb259bd34106e

**Comment by licaon-kter** [comment]
ref: https://gitlab.com/fdroid/fdroiddata/-/jobs/4068758851#L799

**Comment by ivard** [comment]
Thanks for the info. My first impression would be that it is related to a Dart/Flutter pub cache issue or a Java/Gradle version mismatch.

**Comment by licaon-kter** [comment]
You still use Flutter 2 when latest is 3.7.10?

**Comment by ivard** [comment]
Yes, that's correct. We didn't want to combine a major app update with a major Flutter upgrade. That's something we've planned for the following months.

**Comment by licaon-kter** [comment]
The actual issue was a bit above in the CI log: https://gitlab.com/fdroid/fdroiddata/-/jobs/4053212260#L590

Since that lib is not FOSS: https://github.com/juliansteenbakker/mobile_scanner/blob/master/android/build.gradle#L54

Ref: https://mvnrepository.com/artifact/com.google.mlkit/barcode-scanning/17.1.0

**Comment by TheLastProject** [comment]
I can't talk for Flutter as I have no experience with it, but I just wanted to say I've used https://github.com/zxing/zxing/ in my Java Android app Catima for years without ever running into issues on F-Droid (and I found it pretty easy to develop with). So maybe that info can help you find a suitable alternative.

**Comment by licaon-kter** [comment]
Ah, that's fine, we can extract it from there then :+1: 

Ping us when ready so we can update the recipe

**Comment by licaon-kter** [comment]
Done https://gitlab.com/fdroid/fdroiddata/-/commit/f28ac58db89557ad0adfabce8cbaa613a07d9b1a and we're picking flutter version from that script so if you keep it updated F-Droid will build with the correct one

**Comment by ivard** [comment]
Nice, thanks for your effort!

**Comment by licaon-kter** [comment]
Maybe something needs an update? https://gitlab.com/fdroid/fdroiddata/-/jobs/5283460015#L893

**Comment by bobhageman** [comment]
@licaon-kter Thanks for reporting. I've created an MR which fixes the issue, see: https://gitlab.com/fdroid/fdroiddata/-/merge_requests/13867

**Comment by DibranMulder** [comment]
@licaon-kter We will disable MRZ scanning on FDroid asap and will investigate an alternative, is that ok?

**Comment by licaon-kter** [comment]
IS this about QR or barcode? maybe look  at https://gitlab.com/fdroid/wiki/-/wikis/Tips-for-fdroiddata-contributors/FOSS-libs#flutter
